### DISCUSSION TITLE
[Suggestion/Features] Quality Of Life Upgrades to the website

### DISCUSSION THREAD

**Discussion by FoxyLoon** [destruc7i0n/crafting#55] [Ideas]
**Hello, I'm a big fan of the site's simplicity and ease of creating and exporting recipes, as well as helping me understand the new 1.21 recipe format. However, I've noticed that the process of creating recipes is a little slow, due to the fact that you have to do everything manually. I also wanted to suggest a few things to improve the quality of the site. That said, I suggest the following:**

1. The introduction of Dark Mode to the site (for night workers) :)

2. The ability to take an item from the list and just drag it through the empty slots, and with LEFT BUTTON of the MOUSE pressed, it will fill the slots as the cursor passes through with the item (this would help fill the recipe slots faster). This would also replace blocks that would already be in the recipe, removing the ones that were there and automatically adding what I'm picking up and dragging into the given slots.

3. I wish there was a button that cleared the recipe in one go, instead of having to remove each block one by one.

4. It would also be nice if the number of items the recipe would make was permanent in all recipes as I created them, i.e. set an initial value [for example - `“count”: 3`], and that value would stay in all recipes I create (until I manually change it) instead of me having to change the number every time I modify the recipe result.

5. The ability to automate the “name of the exported json file” with block names, for example: If I created the recipe “oak_stairs” which only used “oak_planks” as an ingredient, I would like to just leave the following in the name - `{ingredient}__to__{result}` - and when exporting the json file it would be - `oak_planks__to__oak_stairs`. In turn, if I ONLY changed “`oak_planks`” to “`acacia_planks`” I would get - `acacia_planks__to__oak_stairs` in the name (this would also apply if I only changed the final result). This system could be better worked out and could have more conditions, but I believe it would be a spectacular quality of life, especially for those who wanted several recipes and just want a simple, straightforward name that's easy to understand just by reading it.

6. The ability to add more than one type of item for the same end result, for example, something I can create is this:
```
{
    "type": "crafting_shaped",
    "pattern": [
        "#  ",
        "## ",
        "###"
    ],
    "key": {
        "#": [
            {
                "item": "minecraft:stone_bricks"
            },
            {
                "item": "minecraft:chiseled_stone_bricks"
            },
            {
                "item": "minecraft:cracked_stone_bricks"
            }
        ]
    },
    "result": {
        "id": "minecraft:stone_brick_stairs",
        "count": 4
    }
}
```
In other words, with “`stone_bricks`”, “`chiseled_stone_bricks`” or “`cracked_stone_bricks`”, I could create “`stone_brick_stairs`”. If it were possible to facilitate this on the site, I think it would be great!

7. If you could give some examples of the experience gained with some tables/categories it would be good for those who don't know how it works, for example, how much XP would give 0.1 or 0.5 or 1.0, just so the user has an idea of what to put in that section if they need it. And a button to change the time of the furnace, blast furnace or smoking furnace **to the default** would be ideal, like a quick tip (`200 ticks = 10 sec, 20 ticks = 1 sec`).

8. The ability to copy an item that is already in the recipe, just by pressing SHIFT or CTRL on the keyboard or MIDDLE BUTTON on the mouse [don't forget to leave this written on the site for everyone to know :) ].

9. And finally, the ability to quickly change recipe items with a filter. For example, if in the recipe I have 9 oak_planks in each slot of the crafting table, I would like to be able to quickly change oak_planks to cherry_planks, without having to replace all the blocks again. This is a suggestion that is of little use because if the “2.” suggestion above is added, this suggestion may be of little or no use. However, I suggest it anyway, if there is interest in it.

.
.
.

I wish you a continuation of good work, and many congratulations for your creativity in creating this masterpiece. The site has helped me a lot and I'm very grateful to you! Take care!

**Comment by destruc7i0n** [comment]
Hey, thanks for the detailed write-up. A few of these have been added in the new version: dark mode, a clear recipe button, and the output filename is now generated automatically from the recipe items. The rest are still good ideas and I'll keep them in mind.
